server/controller/src/main/java/ai/starwhale/mlops/domain/job/cache/JobLoader.java

```diff
@@ -51,12 +52,11 @@ public JobLoader(HotJobHolder jobHolder, WatchableTaskFactory watchableTaskFacto
 
     public Job load(@NotNull Job job, Boolean resumePausedOrFailTasks) {
         job.getSteps().forEach(step -> {
-            List<Task> tasks = step.getTasks();
+            List<Task> watchableTasks = watchableTaskFactory.wrapTasks(step.getTasks());
+            step.setTasks(watchableTasks);
             if (resumePausedOrFailTasks) {
-                resumeFrozenTasks(tasks);
+                resumeFrozenTasks(watchableTasks);
             }
-            List<Task> watchableTasks = watchableTaskFactory.wrapTasks(tasks);
-            step.setTasks(watchableTasks);
             scheduleReadyTasks(watchableTasks.parallelStream()
                     .filter(t -> t.getStatus() == TaskStatus.READY)
                     .collect(Collectors.toSet()));
```

Comment:
schedule after `jobHolder.adopt(job)`?
The k8s event may arrive before the job is added to the jobHolder

Reply:
> schedule after `jobHolder.adopt(job)`?
> The k8s event may arrive before the job is added to the jobHolder

yes, you are right. Done.